gfx.h

```diff
@@ -8721,12 +8721,6 @@ class GfxInternal
                 submitPipelineBarriers();
                 break;  // break barrier batches to avoid debug layer warnings
             }
-        if(*buffer.resource_state_ == D3D12_RESOURCE_STATE_INDEX_BUFFER &&  // unbind if active index buffer to prevent debug layer errors
-           buffer_handles_.has_handle(bound_index_buffer_.handle) && buffers_[bound_index_buffer_].resource_ == buffer.resource_)
-            command_list_->IASetIndexBuffer(nullptr);
-        if(*buffer.resource_state_ == D3D12_RESOURCE_STATE_VERTEX_AND_CONSTANT_BUFFER &&    // unbind if active vertex buffer to prevent debug layer errors
-           buffer_handles_.has_handle(bound_vertex_buffer_.handle) && buffers_[bound_vertex_buffer_].resource_ == buffer.resource_)
-            command_list_->IASetVertexBuffers(0, 1, nullptr);
```

Comment:
This was unbinding vertex/index buffers if they were also used as a program parameter. Not having a bound vertex buffer generally breaks rendering so it was removed.
Alternatively we could move binding vertex/index buffers to the program/kernel level within gfx instead of globally setting them as this way we would know if a bound buffer is required or not and could default to unbinding for any kernel launch that doesnt need it.